File: frontend/src/model/UserDTO.ts
export interface UserDTO {
  id: number;
  username: string;
  email: string;
  firstName: string;
  lastName: string;
  token: string | null;
  tokenExpiration: string | null;
  createdAt: string | null;
  updatedAt: string | null;
  role: "STUDENT" | "TEACHER" | "ADMIN";
}

File: backend/src/main/java/com/example/iquiz/dto/UserDto.java
package com.example.iquiz.dto;

import com.example.iquiz.enums.UserRole;

import java.time.LocalDateTime;
import java.util.UUID;

public record UserDto(
        UUID id,
        String username,
        String email,
        String firstName,
        String lastName,
        UserRole role,
        LocalDateTime createdAt
) {
}
